draft-tuexen-tsvwg-rfc4960-errata.xml

```diff
@@ -733,6 +733,79 @@ partial_bytes_acked and cwnd in the congestion avoidance phase.</t>
 </section>
 </section>
 
+<section title='HEARTBEAT ACK and the association error counter'>
+<section title='Description of the Problem'>
+<t>Section 8.1 and Section 8.3 of <xref target="RFC4960"/>
+prescribes that the receiver of a HEARTBEAT ACK must reset the
+association overall error counter. In some circumstances, e.g. when
+a router discards DATA chunks but not HEARTBEAT chunks due to the
+larger size of the DATA chunk, it is better to not clear the
+association error counter by the HEARTBEAT ACK and reset it by the
+SACK only.</t>
+</section>
+<section title='Text Changes to the Document'>
+<figure>
+<artwork>
+---------
+Old text: (Section 8.1)
+---------
+
+The counter shall be reset each time a DATA chunk sent to that peer
+endpoint is acknowledged (by the reception of a SACK) or a HEARTBEAT
+ACK is received from the peer endpoint.
+
+---------
+New text: (Section 8.1)
+---------
+
+The counter shall be reset each time a DATA chunk sent to that peer
+endpoint is acknowledged (by the reception of a SACK). When a
+HEARTBEAT ACK is received from the peer endpoint, the counter should
+also be reset. The receiver of the HEARTBEAT ACK may choose not to
+clear the counter while transmitting data on the association. Such
+choice may be motivated by different sizes of the DATA and HEARTBEAT
+chunks. 
+
+---------
+Old text: (Section 8.3)
+---------
+
+Upon the receipt of the HEARTBEAT ACK, the sender of the HEARTBEAT
+should clear the error counter of the destination transport address
+to which the HEARTBEAT was sent, and mark the destination transport
+address as active if it is not so marked.  The endpoint may
+optionally report to the upper layer when an inactive destination
+address is marked as active due to the reception of the latest
+HEARTBEAT ACK.  The receiver of the HEARTBEAT ACK must also clear the
+association overall error count as well (as defined in Section 8.1).
+
+---------
+New text: (Section 8.3)
+---------
+
+Upon the receipt of the HEARTBEAT ACK, the sender of the HEARTBEAT
+should clear the error counter of the destination transport address
+to which the HEARTBEAT was sent, and mark the destination transport
+address as active if it is not so marked. The receiver of the
+HEARTBEAT ACK may choose not to clear the error counter of the
+destination transport address if there are valid reasons for it, e.g.
+outstanding data on the destination address. The endpoint may
+optionally report to the upper layer when an inactive destination
```

Comment:
On Oct 20, 2015, at 2:57 AM, Maxim <notifications@github.com> wrote:

> In draft-tuexen-tsvwg-rfc4960-errata.xml:
> 
> > +address is marked as active due to the reception of the latest
> > +HEARTBEAT ACK.  The receiver of the HEARTBEAT ACK must also clear the
> > +association overall error count as well (as defined in Section 8.1).
> > +
> > +---------
> > +New text: (Section 8.3)
> > +---------
> > +
> > +Upon the receipt of the HEARTBEAT ACK, the sender of the HEARTBEAT
> > +should clear the error counter of the destination transport address
> > +to which the HEARTBEAT was sent, and mark the destination transport
> > +address as active if it is not so marked. The receiver of the
> > +HEARTBEAT ACK may choose not to clear the error counter of the
> > +destination transport address if there are valid reasons for it, e.g.
> > +outstanding data on the destination address. The endpoint may
> > +optionally report to the upper layer when an inactive destination
> The rule is not changed actually. The RFC uses "should" in respect to the destination address error counter. I think it was used intentionally but then there is a lack of clarity what is motivation. The idea is to just explain motivation
> 

Protocol specifications really should not be in the business of explaining motivation.. or so I have been
told many times over the years :-)

> . I've described above scenarios where it can be useful. In general it is the same as for the association counter: if there is DATA and HB simultaneously then the receiver of HB ACK may decide to not clear the destination error counter as SACK or T3 will provide more exact information about address reachability. I agree that for the destination address it is rarer and less important than for the association as HBs are usually sent to idle addresses. However it is more important if the user initiates on-demand HB sending.
> 
I don’t think we need to add so much description.

R

> —
> Reply to this email directly or view it on GitHub.
> 

--------
Randall Stewart
rrs@netflix.com
803-317-4952